cggmp21/src/signing.rs

```diff
@@ -683,13 +720,32 @@ where
     let K_i = dec_i
         .encrypt_with(&utils::scalar_to_bignumber(&k_i), &rho_i)
         .map_err(|_| Bug::PaillierEnc(BugSource::K_i))?;
+
+    tracer.stage("Generate a_i, b_i, A_i1, A_i2, B_i1, B_i2");
+    let Y_i = {
+        let y_i = SecretScalar::<E>::random(rng);
+        Point::generator() * y_i
+    };
+    let a_i = SecretScalar::<E>::random(rng);
+    let b_i = SecretScalar::<E>::random(rng);
+
+    let A_i1 = &a_i * Point::generator();
+    let A_i2 = &a_i * Y_i + &k_i * Point::generator();
+
+    let B_i1 = &b_i * Point::generator();
+    let B_i2 = &b_i * Y_i + &gamma_i * Point::generator();
     runtime.yield_now().await;
 
     tracer.send_msg();
     outgoings
         .feed(Outgoing::broadcast(Msg::Round1a(MsgRound1a {
             K: K_i.clone(),
             G: G_i.clone(),
+            Y: Y_i,
+            A1: A_i1,
+            A2: A_i2,
+            B1: B_i1,
+            B2: B_i2,
         })))
         .await
         .map_err(IoError::send_message)?;
```

Comment:
Might instead want to yield between the two multiplications. I don't think this yield there is useful as we're going to yield when sending the messages right after that

Reply:
Yeah I think we only need to add yielding in a long computation where there's no `.await`. If there's one (even if it's sending), it should be okay